### ISSUE TITLE
Testing required element after type element

### ISSUE THREAD

**Issue by shenlong3010** [FreeCodeCamp/FreeCodeCamp#9853]
Challenge [Use HTML5 to Require a Field](https://www.freecodecamp.com/challenges/use-html5-to-require-a-field#?solution=%0A%3Clink%20href%3D%22https%3A%2F%2Ffonts.googleapis.com%2Fcss%3Ffamily%3DLobster%22%20rel%3D%22stylesheet%22%20type%3D%22text%2Fcss%22%3E%0A%3Cstyle%3E%0A%20%20.red-text%20%7B%0A%20%20%20%20color%3A%20red%3B%0A%20%20%7D%0A%0A%20%20h2%20%7B%0A%20%20%20%20font-family%3A%20Lobster%2C%20Monospace%3B%0A%20%20%7D%0A%0A%20%20p%20%7B%0A%20%20%20%20font-size%3A%2016px%3B%0A%20%20%20%20font-family%3A%20Monospace%3B%0A%20%20%7D%0A%0A%20%20.thick-green-border%20%7B%0A%20%20%20%20border-color%3A%20green%3B%0A%20%20%20%20border-width%3A%2010px%3B%0A%20%20%20%20border-style%3A%20solid%3B%0A%20%20%20%20border-radius%3A%2050%25%3B%0A%20%20%7D%0A%0A%20%20.smaller-image%20%7B%0A%20%20%20%20width%3A%20100px%3B%0A%20%20%7D%0A%3C%2Fstyle%3E%0A%0A%3Ch2%20class%3D%22red-text%22%3ECatPhotoApp%3C%2Fh2%3E%0A%0A%3Cp%3EClick%20here%20for%20%3Ca%20href%3D%22%23%22%3Ecat%20photos%3C%2Fa%3E.%3C%2Fp%3E%0A%0A%3Ca%20href%3D%22%23%22%3E%3Cimg%20class%3D%22smaller-image%20thick-green-border%22%20alt%3D%22A%20cute%20orange%20cat%20lying%20on%20its%20back%22%20src%3D%22https%3A%2F%2Fbit.ly%2Ffcc-relaxing-cat%22%3E%3C%2Fa%3E%0A%0A%3Cp%3EThings%20cats%20love%3A%3C%2Fp%3E%0A%3Cul%3E%0A%20%20%3Cli%3Ecat%20nip%3C%2Fli%3E%0A%20%20%3Cli%3Elaser%20pointers%3C%2Fli%3E%0A%20%20%3Cli%3Elasagna%3C%2Fli%3E%0A%3C%2Ful%3E%0A%3Cp%3ETop%203%20things%20cats%20hate%3A%3C%2Fp%3E%0A%3Col%3E%0A%20%20%3Cli%3Eflea%20treatment%3C%2Fli%3E%0A%20%20%3Cli%3Ethunder%3C%2Fli%3E%0A%20%20%3Cli%3Eother%20cats%3C%2Fli%3E%0A%3C%2Fol%3E%0A%3Cform%20fccfaa%3D%22%2Fsubmit-cat-photo%22%3E%0A%20%20%3Cinput%20type%3D%22text%22%20required%20placeholder%3D%22cat%20photo%20URL%22%3E%0A%20%20%3Cbutton%20type%3D%22submit%22%3ESubmit%3C%2Fbutton%3E%0A%3C%2Fform%3E%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.82 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

My code:
```html

<link href="https://fonts.googleapis.com/css?family=Lobster" rel="stylesheet" type="text/css">
<style>
  .red-text {
    color: red;
  }

  h2 {
    font-family: Lobster, Monospace;
  }

  p {
    font-size: 16px;
    font-family: Monospace;
  }

  .thick-green-border {
    border-color: green;
    border-width: 10px;
    border-style: solid;
    border-radius: 50%;
  }

  .smaller-image {
    width: 100px;
  }
</style>

<h2 class="red-text">CatPhotoApp</h2>

<p>Click here for <a href="#">cat photos</a>.</p>

<a href="#"><img class="smaller-image thick-green-border" alt="A cute orange cat lying on its back" src="https://bit.ly/fcc-relaxing-cat"></a>

<p>Things cats love:</p>
<ul>
  <li>cat nip</li>
  <li>laser pointers</li>
  <li>lasagna</li>
</ul>
<p>Top 3 things cats hate:</p>
<ol>
  <li>flea treatment</li>
  <li>thunder</li>
  <li>other cats</li>
</ol>
<form action="/submit-cat-photo">
  <input type="text" required placeholder="cat photo URL">
  <button type="submit">Submit</button>
</form>

```

It seems that the required input doesn't work as it should be. I still can proceed to my next challenge even though I didn't put any in the requirement box.
Appreciate if you look at it.

Thanks.

**Comment by raisedadead** [comment]
@shenlong3010 

Did you visit anyone else's challenge and submit it, or have you completed this already?
looking at the code you shared above, a snippet:
```
<form action="/submit-cat-photo">
  <input type="text" required placeholder="cat photo URL"> //<-------------------- here
  <button type="submit">Submit</button>
</form>
```

You have the attribute `require` in your code.
Try resetting the code from the reset button.

**Comment by BKinahan** [comment]
I think what @shenlong3010 may be trying to indicate is that since the aim of the challenge is to create a text input field which requires text to be entered before the form can be submitted, they interpret this to mean that the tests should not be passed if the user does not enter text into the mobile preview before running the tests.

This could be a misunderstanding about the nature and purpose of the `required` attribute and its relation to the challenge scenario. It functions correctly in the code above because clicking the `Submit` button without text in the field prompts the user to enter text, and clicking it with text submits the form correctly and displays a "Submitted!" message. A user may test this before submitting their correct code, but the tests do not force them to do so.

**Comment by raisedadead** [comment]
I am really uncertain though how we can test the elements on the iPhone preview. 
Currently there is no implementation that will let us do that.

Our tests are basically unit tests on the code entered in the editor.

**Comment by BKinahan** [comment]
Yes, I think it's not feasible or desirable to test those elements, and it's better leave it to users to test for themselves if they wish.

I believe it was just a small misunderstanding about the word "required" in this case. Though I find the instructions quite clear, there's always the perhaps unavoidable possibility of language issues for people new to learning code or non-native English speakers.

**Comment by shenlong3010** [comment]
The challenge is easy, so I passed it anyway. As I mentioned, the
"required" elements is easy to understand, but to truly be understandable,
not only for me also for other new learners, there should be a "required
input"​.
I tried twice, first is with some text, second is without the text. All of
them seem "pass the challenge".
It's not a big deal, but just to remind you guys, that's all.
Thanks for replying back.

Sincere.

Loc.
